1.4.0 is linting ignored files

ESLint just upgraded and now our Travis builds are failing because it's linting an ignored file.

Travis build: https://travis-ci.org/discourse/discourse/jobs/79870981
It runs `eslint app/assets/javascripts` from the project directory.

The `.eslintignore` file is in the project directory: https://github.com/discourse/discourse/blob/22844b9e4663a1dd06040cd5913f02f571f9672f/.eslintignore and contains:
```
...
app/assets/javascripts/locales/i18n.js
app/assets/javascripts/defer/html-sanitizer-bundle.js
app/assets/javascripts/discourse/lib/Markdown.Editor.js
...
```

But the build outputs:

```
/home/travis/build/discourse/discourse/app/assets/javascripts/defer/html-sanitizer-bundle.js
  1243:14  error  Octal literal in strict mode
```

This happens on my local machine too.
```
$ eslint --version
v1.4.0
```

This just started happening with the new release (a few minutes ago).

Adding `--ignore-path` does not help.

```
$ eslint --ignore-path ./.eslintignore app/assets/javascripts/

/home/kane/projects/discourse/app/assets/javascripts/defer/html-sanitizer-bundle.js
  1243:14  error  Octal literal in strict mode
```

--- a/lib/cli-engine.js
+++ b/lib/cli-engine.js
@@ -329,7 +329,7 @@ function processPath(extensions) {
             newPath = pathname + suffix;
         }
 
-        return newPath.replace(/\\/g, "/");
+        return newPath.replace(/\\/g, "/").replace(/^\.\//, "");
     };
 }
 
@@ -493,7 +493,6 @@ CLIEngine.prototype = {
 
         globOptions = {
             nodir: true,
-            realpath: true,
             ignore: ignoredPathsList
         };
 
@@ -504,6 +503,8 @@ CLIEngine.prototype = {
          * @returns {void}
          */
         function executeOnFile(filename) {
+            filename = fs.realpathSync(filename);
+
             if (processed[filename]) {
                 return;
             }
@@ -559,7 +560,7 @@ CLIEngine.prototype = {
 
         patterns.forEach(function(pattern) {
             if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
-                executeOnFile(fs.realpathSync(pattern));
+                executeOnFile(pattern);
             } else {
                 glob.sync(pattern, globOptions).forEach(executeOnFile);
             }

--- a/tests/lib/cli-engine.js
+++ b/tests/lib/cli-engine.js
@@ -460,6 +460,24 @@ describe("CLIEngine", function() {
             assert.equal(report.results.length, 0);
         });
 
+        it("should return zero messages when all given files are ignored event with a `./` prefix", function() {
+            engine = new CLIEngine({
+                ignorePath: getFixturePath(".eslintignore")
+            });
+
+            var report = engine.executeOnFiles(["./tests/fixtures/"]);
+            assert.equal(report.results.length, 0);
+        });
+
+        it("should return zero messages when all given files are ignored via ignore-pattern", function() {
+            engine = new CLIEngine({
+                ignorePattern: "tests/fixtures/single-quoted.js"
+            });
+
+            var report = engine.executeOnFiles(["tests/fixtures/*-quoted.js"]);
+            assert.equal(report.results.length, 0);
+        });
+
         it("should return a warning when an explicitly given file is ignored", function() {
             engine = new CLIEngine({
                 ignorePath: getFixturePath(".eslintignore")

